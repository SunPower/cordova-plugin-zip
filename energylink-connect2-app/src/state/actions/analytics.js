import { namedAction } from 'shared/redux-utils'

const analyticsAction = namedAction('Analytics')
const mixpanelActions = namedAction('Mixpanel')

export const BEGIN_INSTALL = analyticsAction('BEGIN_INSTALL')
export const COMMISSIONING_FINISH = analyticsAction('FINISH COMMISSIONING')
export const COMMISSION_SUCCESS = analyticsAction('COMMISSION_SUCCESS')
export const CONFIG_START = analyticsAction('Configure start')
export const CONNECT_PVS_CAMERA = analyticsAction('CONNECT_PVS_CAMERA')
export const CONNECT_PVS_MANUALLY = analyticsAction('CONNECT_PVS_MANUALLY')
export const FINISH_PLT_SETUP = analyticsAction('Finish PLT setup')
export const MIXPANEL_EVENT_ERROR = analyticsAction('MIXPANEL_EVENT_ERROR')
export const MIXPANEL_EVENT_QUEUED = mixpanelActions('Event queued')
export const RESET_PVS_INTERNET_TRACKING = analyticsAction(
  'RESET_PVS_INTERNET_TRACKING'
)
export const SCANNING_START = analyticsAction('Scanning start')
export const SET_AC_DEVICES = analyticsAction('Set AC modules')
export const CLAIM_DEVICES_MIXPANEL_EVENT = analyticsAction(
  'CLAIM DEVICES MIXPANEL EVENT'
)
export const START_BULK_SETTINGS_TIMER = analyticsAction('Start bulk settings')
export const START_PLT_SETUP = analyticsAction('Start PLT setup')
