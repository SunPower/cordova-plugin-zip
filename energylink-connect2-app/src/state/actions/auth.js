import { createAction } from 'redux-act'
import {
  pipe,
  equals,
  head,
  last,
  prop,
  merge,
  omit,
  identity,
  converge,
  useWith
} from 'ramda'
import { getFile } from './fileDownloader'
import authClient from 'shared/auth/sdk'
import { httpGet } from 'shared/fetch'

export const LOGIN_INIT = createAction('LOGIN_INIT')
export const LOGIN_SUCCESS = createAction('LOGIN_SUCCESS')
export const LOGIN_ERROR = createAction('LOGIN_ERROR')
export const LOGOUT = createAction('LOGOUT')

const ROLES = {
  PARTNER: 'partner',
  PARTNER_PRO: 'partner_pro'
}

export const requestLogin = () => {
  return (dispatch, state) => {
    try {
      let state = authClient.generateRandomValue()
      dispatch(LOGIN_INIT({ state }))
      authClient.authorizeOAuth(state)
    } catch (err) {
      dispatch(LOGIN_ERROR({ message: err }))
    }
  }
}

export const handleLoginFromPing = URL => {
  return (dispatch, getState) => {
    const { user } = getState()
    const hashes = authClient.parseURL(URL)

    if (hashes.error && hashes.error_description) {
      const error = `${hashes.error}: ${hashes.error_description}`
      dispatch(LOGIN_ERROR({ message: error }))
      return
    }

    if (hashes.state && !equals(hashes.state, user.auth.state)) {
      dispatch(LOGIN_ERROR({ message: 'STATE_MISMATCH' }))
      return
    }

    if (hashes.code) {
      authClient
        .getAccessTokenOAuth(hashes.code)
        .then(token => {
          dispatch(handleUserProfile(token))
        })
        .catch(error => {
          dispatch(LOGIN_ERROR({ message: 'FETCH_ACCESS_TOKEN' }))
        })
    }
  }
}

const discardUserPropsFromAPIResponse = pipe(
  prop('data'),
  omit(['userId', 'displayName', 'email'])
)

const merge_EDP_PING_Response = useWith(merge, [
  discardUserPropsFromAPIResponse,
  identity
])

const buildUser = converge(merge_EDP_PING_Response, [head, last])

export const handleUserProfile = (tokenInfo = {}) => {
  return dispatch => {
    try {
      const { access_token } = tokenInfo
      Promise.all([
        httpGet('/auth/user', null, access_token),
        authClient.getUserInfoOAuth(access_token)
      ])
        .then(buildUser)
        .then(user => {
          const payload = { data: user, auth: tokenInfo }
          const ug = user.userGroup.toLowerCase()
          if (ug !== ROLES.PARTNER && ug !== ROLES.PARTNER_PRO) {
            dispatch(LOGIN_ERROR({ message: 'INVALID_ROLE' }))
          } else {
            dispatch(LOGIN_SUCCESS(payload))
            //dispatch(fetchInventory())
            dispatch(getFile())
          }
        })
        .catch(error => {
          console.error(error)
          const details = prop('details', error)

          if (!details) {
            dispatch(LOGIN_ERROR({ message: 'HTTP_ERROR' }))
            return
          }

          const firstError = head(details)
          const firstErrorCode = prop('code', firstError)

          if (equals(firstErrorCode, 'INVALID_VALUE')) {
            const firstErrorMsg = prop('message', firstError)
            const ERROR = firstErrorMsg.includes('Access token expired')
              ? 'ACCESS_TOKEN_EXPIRED'
              : firstErrorMsg
            dispatch(LOGIN_ERROR({ message: ERROR }))
          } else {
            const description = prop('error_description', error)
            dispatch(LOGIN_ERROR({ message: description }))
          }
        })
    } catch (err) {
      dispatch(LOGIN_ERROR(err))
    }
  }
}

export const verifyToken = access_token => {
  return dispatch => {
    authClient
      .verifyTokenOAuth(access_token)
      .then(response => {
        if (!response.uniqueId) {
          dispatch(LOGIN_ERROR({ message: 'ACCESS_TOKEN_EXPIRED' }))
          dispatch(LOGOUT())
          return
        }
        dispatch(VALIDATE_SESSION_SUCCESS())
      })
      .catch(error => {
        dispatch(LOGIN_ERROR({ message: 'ACCESS_TOKEN_EXPIRED' }))
        dispatch(LOGOUT())
        dispatch(VALIDATE_SESSION_ERROR())
      })
  }
}

export const VALIDATE_SESSION_INIT = createAction('VALIDATE_SESSION_INIT')
export const VALIDATE_SESSION_SUCCESS = createAction('VALIDATE_SESSION_SUCCESS')
export const VALIDATE_SESSION_ERROR = createAction('VALIDATE_SESSION_ERROR')

export const validateSession = () => {
  return (dispatch, getState) => {
    const { user } = getState()
    if (user.auth.access_token) {
      dispatch(VALIDATE_SESSION_INIT())
      dispatch(verifyToken(user.auth.access_token))
    }
  }
}
