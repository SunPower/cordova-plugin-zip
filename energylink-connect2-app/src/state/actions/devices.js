import { createAction } from 'redux-act'

export const DISCOVER_INIT = createAction('discover devices init')
export const DISCOVER_UPDATE = createAction('discover devices update')
export const DISCOVER_COMPLETE = createAction('discover devices complete')
export const DISCOVER_ERROR = createAction('discover devices error')
export const PUSH_CANDIDATES_INIT = createAction('PUSH_CANDIDATES_INIT')
export const PUSH_CANDIDATES_SUCCESS = createAction('PUSH_CANDIDATES_SUCCESS')
export const PUSH_CANDIDATES_ERROR = createAction('PUSH_CANDIDATES_ERROR')
export const FETCH_CANDIDATES_INIT = createAction('FETCH_CANDIDATES_INIT')
export const FETCH_CANDIDATES_UPDATE = createAction('FETCH_CANDIDATES_UPDATE')
export const FETCH_CANDIDATES_ERROR = createAction('FETCH_CANDIDATES_ERROR')
export const FETCH_CANDIDATES_COMPLETE = createAction(
  'FETCH_CANDIDATES_COMPLETE'
)
export const CLAIM_DEVICES_INIT = createAction('CLAIM_DEVICES_INIT')
export const CLAIM_DEVICES_SUCCESS = createAction('CLAIM_DEVICES_SUCCESS')
export const CLAIM_DEVICES_ERROR = createAction('CLAIM_DEVICES_ERROR')
export const RESET_DISCOVERY = createAction('RESET_DISCOVERY')
