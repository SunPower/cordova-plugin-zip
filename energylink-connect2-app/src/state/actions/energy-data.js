import { createAction } from 'redux-act'

export const FETCH_LTE_DATA_INIT = createAction('FETCH_LTE_DATA_INIT')
export const FETCH_LTE_DATA_SUCCESS = createAction('FETCH_LTE_DATA_SUCCESS')
export const FETCH_LTE_DATA_ERROR = createAction('FETCH_LTE_DATA_ERROR')

export const FETCH_CURRENT_POWER_INIT = createAction('FETCH_CURRENT_POWER_INIT')
export const FETCH_CURRENT_POWER_SUCCESS = createAction(
  'FETCH_CURRENT_POWER_SUCCESS'
)
export const FETCH_CURRENT_POWER_ERROR = createAction(
  'FETCH_CURRENT_POWER_ERROR'
)

export const GET_ENERGY_DATA_INIT = createAction('GET_ENERGY_DATA_INIT')
export const GET_ENERGY_DATA_SUCCESS = createAction('GET_ENERGY_DATA_SUCCESS')
export const GET_ENERGY_DATA_ERROR = createAction('GET_ENERGY_DATA_ERROR')

export const LIVE_ENERGY_DATA_NOTIFICATION = createAction(
  'LIVE_ENERGY_DATA_NOTIFICATION'
)
export const LIVE_ENERGY_DATA_DAILY = createAction('LIVE_ENERGY_DATA_DAILY')
export const LIVE_ENERGY_DATA_CONNECTION = createAction(
  'LIVE_ENERGY_DATA_CONNECTION'
)
export const LIVE_ENERGY_DATA_ERROR = createAction('LIVE_ENERGY_DATA_ERROR')

export const GET_POWER_DATA_INIT = createAction('GET_POWER_DATA_INIT')
export const GET_POWER_DATA_SUCCESS = createAction('GET_POWER_DATA_SUCCESS')
export const GET_POWER_DATA_ERROR = createAction('GET_POWER_DATA_ERROR')

export const INTERVALS = {
  FIVE_MINUTE: 'five_minute',
  QUARTER_HOUR: 'quarter_hour',
  HOUR: 'hour',
  DAY: 'day',
  MONTH: 'month',
  YEAR: 'year',
  LIVE: 'live'
}

// Minimum amount of time that has to pass before fetching more data
export const DELTAS = {
  [INTERVALS.HOUR]: 1000 * 60 * 5 /* 5 minutes */,
  [INTERVALS.DAY]: 1000 * 60 * 30 /* 30 minutes */,
  [INTERVALS.MONTH]: 1000 * 60 * 60 * 12 /* 12 hours */,
  [INTERVALS.YEAR]: 1000 * 60 * 60 * 24 * 7 /* 1 week */
}

export const ENERGY_DATA_START_POLLING = createAction(
  'ENERGY_DATA_START_POLLING'
)
export const ENERGY_DATA_POLLING = createAction('ENERGY_DATA_POLLING')
export const ENERGY_DATA_STOP_POLLING = createAction('ENERGY_DATA_STOP_POLLING')

export const POWER_DATA_START_POLLING = createAction('POWER_DATA_START_POLLING')
export const POWER_DATA_POLLING = createAction('POWER_DATA_POLLING')
export const POWER_DATA_STOP_POLLING = createAction('POWER_DATA_STOP_POLLING')

export const CURRENT_POWER_DATA_START_POLLING = createAction(
  'CURRENT_POWER_DATA_START_POLLING'
)
export const CURRENT_POWER_DATA_STOP_POLLING = createAction(
  'CURRENT_POWER_DATA_STOP_POLLING'
)

export const NORMALIZE_ENERGY_DATA_INIT = createAction(
  'NORMALIZE_ENERGY_DATA_INIT'
)
export const NORMALIZE_ENERGY_DATA_SUCCESS = createAction(
  'NORMALIZE_ENERGY_DATA_SUCCESS'
)

export const NORMALIZE_POWER_DATA_INIT = createAction(
  'NORMALIZE_POWER_DATA_INIT'
)
export const NORMALIZE_POWER_DATA_SUCCESS = createAction(
  'NORMALIZE_POWER_DATA_SUCCESS'
)

export const LTE_DATA_POLL_START = createAction('LTE_DATA_POLL_START')
export const LTE_DATA_POLL_STOP = createAction('LTE_DATA_POLL_STOP')
