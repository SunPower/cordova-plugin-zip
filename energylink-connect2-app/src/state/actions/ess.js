import { createAction } from 'redux-act'

export const DOWNLOAD_META_INIT = createAction('DOWNLOAD_META_INIT')
export const DOWNLOAD_META_SUCCESS = createAction('DOWNLOAD_META_SUCCESS')
export const DOWNLOAD_META_ERROR = createAction('DOWNLOAD_META_ERROR')

export const DOWNLOAD_OS_INIT = createAction('DOWNLOAD_OS_INIT')
export const DOWNLOAD_OS_SUCCESS = createAction('DOWNLOAD_OS_SUCCESS')
export const DOWNLOAD_OS_PROGRESS = createAction('DOWNLOAD_OS_PROGRESS')
export const DOWNLOAD_OS_ERROR = createAction('DOWNLOAD_OS_ERROR')
