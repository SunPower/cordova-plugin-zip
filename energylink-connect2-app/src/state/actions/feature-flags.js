import { createAction } from 'redux-act'

export const UPDATE_FEATURE_FLAGS = createAction('UPDATE_FEATURE_FLAGS')
export const UPDATE_FEATURE_FLAGS_ERROR = createAction(
  'UPDATE_FEATURE_FLAGS_ERROR'
)
