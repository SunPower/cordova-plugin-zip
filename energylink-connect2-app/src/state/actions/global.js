import { createAction } from 'redux-act'

export const SET_LAST_VISITED_PAGE = createAction('SET_LAST_VISITED_PAGE')
export const RESET_LAST_VISITED_PAGE = createAction('RESET_LAST_VISITED_PAGE')
export const CHECK_APP_UPDATE_INIT = createAction('CHECK_APP_UPDATE_INIT')
export const CHECK_APP_UPDATE_SUCCESS = createAction('CHECK_APP_UPDATE_SUCCESS')
export const CHECK_APP_UPDATE_ERROR = createAction('CHECK_APP_UPDATE_ERROR')
export const APP_UPDATE_OPEN_MARKET = createAction('APP_UPDATE_OPEN_MARKET')
export const APP_UPDATE_OPEN_MARKET_SUCCESS = createAction(
  'APP_UPDATE_OPEN_MARKET_SUCCESS'
)
export const APP_UPDATE_OPEN_MARKET_ERROR = createAction(
  'APP_UPDATE_OPEN_MARKET_ERROR'
)
export const RESET_COMMISSIONING = createAction('RESET_COMMISSIONING')
export const CHECK_SSL_CERTS = createAction('CHECK_SSL_CERTS')
export const CHECK_SSL_CERTS_SUCCESS = createAction('CHECK_SSL_CERTS_SUCCESS')
export const CHECK_SSL_CERTS_ERROR = createAction('CHECK_SSL_CERTS_ERROR')
