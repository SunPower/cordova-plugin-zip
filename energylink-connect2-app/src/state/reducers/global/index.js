import { createReducer } from 'redux-act'
import {
  RESET_FEEDBACK_FORM,
  SEND_FEEDBACK_ERROR,
  SEND_FEEDBACK_INIT,
  SEND_FEEDBACK_SUCCESS
} from '../../actions/feedback'
import { DEVICE_RESUME } from '../../actions/mobile'
import {
  DATA_SOURCES,
  GRAPHS,
  SELECT_DATA_SOURCE,
  SELECT_ENERGY_GRAPH
} from '../../actions/user'
import {
  SET_LAST_VISITED_PAGE,
  RESET_LAST_VISITED_PAGE,
  CHECK_APP_UPDATE_SUCCESS,
  CHECK_APP_UPDATE_ERROR
} from 'state/actions/global'
import paths from 'routes/paths'
import { SET_SCANDIT_ACCESS } from 'state/actions/scandit'
import { RESET_COMMISSIONING } from 'state/actions/global'

const initialState = {
  isAccountCreated: false,
  isDeviceResumeListened: false,
  selectedEnergyGraph: GRAPHS.POWER,
  isSendingFeedback: false,
  isFeedbackSuccessful: false,
  selectedDataSource: DATA_SOURCES.LIVE,
  feedbackError: null,
  lastVisitedPage: paths.PROTECTED.PVS_SELECTION_SCREEN.path,
  canAccessScandit: true,
  updateAvailable: false,
  updateVersion: 0
}

export const globalReducer = createReducer(
  {
    [SELECT_ENERGY_GRAPH]: (state, action) => ({
      ...state,
      selectedEnergyGraph: action
    }),
    [SELECT_DATA_SOURCE]: (state, action) => ({
      ...state,
      selectedDataSource: action
    }),
    [DEVICE_RESUME]: state => ({
      ...state,
      isDeviceResumeListened: true
    }),
    [SEND_FEEDBACK_INIT]: state => ({
      ...state,
      isSendingFeedback: true
    }),
    [SEND_FEEDBACK_SUCCESS]: state => ({
      ...state,
      isSendingFeedback: false,
      isFeedbackSuccessful: true,
      feedbackError: false
    }),
    [SEND_FEEDBACK_ERROR]: (state, payload) => ({
      ...state,
      feedbackError: payload,
      isSendingFeedback: false
    }),
    [RESET_FEEDBACK_FORM]: state => ({
      ...state,
      isFeedbackSuccessful: false
    }),
    [SET_LAST_VISITED_PAGE]: (state, lastVisitedPage) => ({
      ...state,
      lastVisitedPage
    }),
    [RESET_LAST_VISITED_PAGE]: state => ({
      ...state,
      lastVisitedPage: initialState.lastVisitedPage
    }),
    [RESET_COMMISSIONING]: () => initialState,
    [SET_SCANDIT_ACCESS]: (state, canAccessScandit) => ({
      ...state,
      canAccessScandit
    }),
    [CHECK_APP_UPDATE_SUCCESS]: (state, updateVersion) => ({
      ...state,
      updateAvailable: true,
      updateVersion
    }),
    [CHECK_APP_UPDATE_ERROR]: state => ({
      ...state,
      updateAvailable: false,
      updateVersion: 0
    })
  },
  initialState
)
