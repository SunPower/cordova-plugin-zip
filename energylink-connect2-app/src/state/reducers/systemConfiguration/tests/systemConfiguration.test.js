import {
  submitConfigReducer,
  initialState,
  preconfigStates
} from '../submitConfiguration'

import { RESET_COMMISSIONING } from 'state/actions/global'
import {
  SUBMIT_CLEAR,
  REPLACE_RMA_PVS,
  SUBMIT_CONFIG,
  SUBMIT_CONFIG_SUCCESS,
  SUBMIT_CONFIG_ERROR,
  SUBMIT_COMMISSION_SUCCESS,
  SUBMIT_COMMISSION_ERROR,
  RESET_SYSTEM_CONFIGURATION,
  ALLOW_COMMISSIONING,
  SUBMIT_PRECONFIG_GRIDPROFILE,
  SUBMIT_PRECONFIG_ERROR,
  SUBMIT_PRECONFIG_SUCCESS
} from 'state/actions/systemConfiguration'

describe('[Reducer] System Configuration', function() {
  it('should have the correct actions', function() {
    expect(submitConfigReducer.has(SUBMIT_CLEAR)).toBe(true)
    expect(submitConfigReducer.has(REPLACE_RMA_PVS)).toBe(true)
    expect(submitConfigReducer.has(SUBMIT_CONFIG)).toBe(true)
    expect(submitConfigReducer.has(SUBMIT_CONFIG_SUCCESS)).toBe(true)
    expect(submitConfigReducer.has(SUBMIT_CONFIG_ERROR)).toBe(true)
    expect(submitConfigReducer.has(SUBMIT_COMMISSION_SUCCESS)).toBe(true)
    expect(submitConfigReducer.has(SUBMIT_COMMISSION_ERROR)).toBe(true)
    expect(submitConfigReducer.has(RESET_SYSTEM_CONFIGURATION)).toBe(true)
    expect(submitConfigReducer.has(ALLOW_COMMISSIONING)).toBe(true)
    expect(submitConfigReducer.has(SUBMIT_PRECONFIG_GRIDPROFILE)).toBe(true)
    expect(submitConfigReducer.has(SUBMIT_PRECONFIG_ERROR)).toBe(true)
    expect(submitConfigReducer.has(SUBMIT_PRECONFIG_SUCCESS)).toBe(true)
    expect(submitConfigReducer.has(RESET_COMMISSIONING)).toBe(true)
  })

  it('should not change the initial state', function() {
    expect(initialState).toMatchSnapshot()
  })

  it('should not change the state if no action is passed', function() {
    expect(submitConfigReducer(undefined, {})).toEqual(initialState)
  })

  it('should set submitting to true and config to a valid config object on SUBMIT_CONFIG', function() {
    const { submitting, config, ...initialRest } = initialState
    const configPayload = {
      site: 'a'
    }
    const {
      submitting: submittingReduced,
      config: configReduced,
      ...reducedRest
    } = submitConfigReducer(undefined, SUBMIT_CONFIG(configPayload))
    expect(submittingReduced).toBe(true)
    expect(configReduced).toStrictEqual(configPayload)
    expect(initialRest).toStrictEqual(reducedRest)
    expect(submitting).not.toEqual(submittingReduced)
    expect(config).not.toEqual(configReduced)
  })

  it('should set preconfigState to STARTED as soon as we submit grid profile', function() {
    const { preconfigState, ...initialRest } = initialState
    const payload = {
      gridProfile: {}
    }
    const {
      preconfigState: preconfigStateReduced,
      ...reducedRest
    } = submitConfigReducer(undefined, SUBMIT_PRECONFIG_GRIDPROFILE(payload))

    expect(preconfigStateReduced).toBe(preconfigStates.STARTED)
    expect(initialRest).toStrictEqual(reducedRest)
  })

  it('should set preconfigState to ERROR and set a message in preconfigError if SUBMIT_PRECONFIG_ERROR is dispatched', function() {
    const { preconfigState, preconfigError, ...initialRest } = initialState
    const errorMessage = 'Something has failed'
    const {
      preconfigState: preconfigStateReduced,
      preconfigError: preconfigErrorReduced,
      ...reducedRest
    } = submitConfigReducer(undefined, SUBMIT_PRECONFIG_ERROR(errorMessage))

    expect(preconfigStateReduced).toBe(preconfigStates.ERROR)
    expect(preconfigErrorReduced).toBe(errorMessage)
    expect(initialRest).toStrictEqual(reducedRest)
  })

  it('should set preconfigState to SUCCESS and clear error message if SUBMIT_PRECONFIG_SUCCESS is dispatched', function() {
    const { preconfigState, preconfigError, ...initialRest } = initialState
    const {
      preconfigState: preconfigStateReduced,
      preconfigError: preconfigErrorReduced,
      ...reducedRest
    } = submitConfigReducer(undefined, SUBMIT_PRECONFIG_SUCCESS({}))

    expect(preconfigStateReduced).toBe(preconfigStates.SUCCESS)
    expect(preconfigErrorReduced).toBe(preconfigError)
    expect(initialRest).toStrictEqual(reducedRest)
  })
})
