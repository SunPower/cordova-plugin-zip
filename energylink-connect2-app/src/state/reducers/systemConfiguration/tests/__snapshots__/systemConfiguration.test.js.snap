// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Reducer] System Configuration should not change the initial state 1`] = `
Object {
  "canCommission": false,
  "commissioned": false,
  "config": Object {},
  "error": "",
  "preconfigError": "",
  "preconfigState": "NOT_STARTED",
  "submitted": false,
  "submitting": false,
}
`;
